[docs] Add EAS Metadata docs

byCedric opened the pull request:
# Why

This is a draft to document the usage of EAS Metadata

# How

Added docs

# Test Plan

See docs: http://localhost:3002/eas-metadata/

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

byCedric committed:
[docs] Add metadataPath to eas json submit schema

byCedric committed:
[docs] Add eas metadata draft pages

byCedric committed:
[docs] Add eas metadata server side redirect from root

byCedric committed:
[docs] Add eas metadata root boxlink

EvanBacon commented on docs/pages/eas-metadata/getting-started.md:
```diff
@@ -0,0 +1,70 @@
+
+## Prerequisites
+
+EAS Metadata is available starting from EAS CLI >= 0.54.0, and _only supports the App Store_.
```
```suggestion
EAS Metadata is available starting from EAS CLI >= 0.54.0, and _currently_ only supports the Apple App Store.
```

EvanBacon commented on docs/pages/eas-metadata/getting-started.md:
```diff
@@ -0,0 +1,70 @@
+
+## Create a local store configuration
+
+We first need to create our local **store.config.json** to get started. This file keeps track of all the information for the app stores. If you have an existing app in the stores, you can generate the configuration file by running:
```
```suggestion
To get started, create a **store.config.json** in the root project directory of your app. This file keeps track of all the information for the app stores. If you have an existing app in the stores, you can generate the configuration file by running:
```

brentvatne commented on docs/pages/eas-metadata/getting-started.md:
```diff
@@ -0,0 +1,70 @@
+
+To create a new app version in the stores, you must upload a new binary. We can do this by running:
+
+<Terminal cmd={[
+  '# Build and deploy to the stores',
+  '$ eas build --auto-submit',
+  '# Or deploy a previous build to the stores',
+  '$ eas submit',
+]} />
```
ideally we can leave this terminal bit out and just link to some other docs page that tells people about getting their app on the store, or assume that they know how to do this already

brentvatne replied:
testing a new tool

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/5bf29d2c-23ae-4af8-8ccc-fe767cd2df79/thread/ef2c770e-87b7-4d63-8858-dc70e3c51f4a?message=611dfd65-7825-4d07-8daa-61f09335b64e).

byCedric replied:
We can link to this if that makes more sense? https://docs.expo.dev/distribution/uploading-apps/

brentvatne replied:
that works!

--

Comment posted using [Unblocked](https://getunblocked.com/dashboard/team/5bf29d2c-23ae-4af8-8ccc-fe767cd2df79/thread/ef2c770e-87b7-4d63-8858-dc70e3c51f4a?message=42fca110-5457-4386-9424-5f204e7168c4).

byCedric replied:
Done!

brentvatne commented on docs/pages/eas-metadata/introduction.md:
```diff
@@ -0,0 +1,19 @@
+
+**EAS Metadata** is a service to help you provide all necessary information to the stores and get your app published.
+
+To get your app into the hands of your users, you have to publish the app on multiple app stores. During this process, you'll have to answer questions about complex topics that often don't apply to your app. After submitting the information, you can start the lengthy review process. If the reviewer finds any issues in the information you sent, you'll have to restart the review process, making the ...
```
```suggestion
To get your app into the hands of your users, you have to publish the app on multiple app stores. During this process, you'll have to answer questions about complex topics that often don't apply to your app. After submitting the information, you can start the lengthy review process. If the reviewer finds any issues in the information you sent, you'll have to restart the review process.
```

byCedric committed:
[docs] Apply feedback on EAS metadata docs

Co-authored-by: Brent Vatne <brentvatne@gmail.com>
Co-authored-by: Evan Bacon <bacon@expo.io>

byCedric committed:
[docs] Drop references to the store config doc

With these changes, we can release it as is, and finalize the store config overview later.

amandeepmittal commented on docs/pages/eas-metadata/getting-started.md:
```diff
@@ -0,0 +1,71 @@
+
+EAS Metadata is available starting from EAS CLI >= 0.54.0, and _currently_ only supports the Apple App Store.
+
+If you are using vscode, also make sure to [install the Expo plugin](https://github.com/expo/vscode-expo) for **store.config.json** auto-completion.
```
```suggestion
If you are using VSCode, make sure to [install the Expo plugin](https://github.com/expo/vscode-expo) for **store.config.json** auto-completion.
```

Let's use the branding name for VSCode.

byCedric replied:
👍

amandeepmittal commented on docs/pages/eas-metadata/getting-started.md:
```diff
@@ -0,0 +1,71 @@
+
+## Update the store config
+
+Now it's time to edit the **store.config.json** file and customize it to your app needs. A complete overview of what can be defined is coming soon. In the meantime, you can use the [vscode Expo plugin](https://github.com/expo/vscode-expo) and use the auto-completion.
```
```suggestion
Now it's time to edit the **store.config.json** file and customize it to your app needs. A complete overview of what can be defined is coming soon. In the meantime, you can use the [VSCode Expo plugin](https://github.com/expo/vscode-expo) and use the auto-completion.
```

amandeepmittal commented on docs/pages/eas-metadata/getting-started.md:
```diff
@@ -0,0 +1,71 @@
+
+## Upload a new app version
+
+Before we can upload the **store.config.json**, you must upload a new binary of your app. You can [read more about uploading new binaries to stores here.](../distribution/uploading-apps.md).
```
```suggestion
Before we can upload the **store.config.json**, you must upload a new binary of your app. You can [read more about uploading new binaries to stores here.](../distribution/uploading-apps.md).
```
```suggestion
Before you can upload the **store.config.json**, you must upload a new binary of your app. [Read more about uploading new binaries to stores](../distribution/uploading-apps.md).
```

As per our Docs Writing Style, let's stick to [writing in second person](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#write-in-the-second-person).

amandeepmittal commented on docs/pages/eas-metadata/getting-started.md:
```diff
@@ -0,0 +1,71 @@
+
+## Upload the store config
+
+When you are happy with the **store.config.json** settings, we can start syncing that data to the app stores. All we need to do is one command:
```
```suggestion
When you are happy with the **store.config.json** settings, start syncing that data to the app stores. All you need to do is run the following command:
```

amandeepmittal reviewed (commented):
Made some small suggestions as per our Writing Style guide.

byCedric reviewed (commented):
woops

byCedric committed:
[docs] Use VSCode over vscode

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

byCedric committed:
[docs] Use second person style

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

byCedric committed:
[docs] Use VS Code instead of VSCode

Simek reviewed (approved):
Great work Cedric! 🙏
